sell-loot.lic

```diff
@@ -19,24 +19,45 @@ class SellLoot
 
     settings = get_settings
     town_data = get_data('town')
-    @actual_fucking_hometown = settings.hometown
+    @character_hometown = settings.hometown
     @hometown = town_data[settings.hometown]
     @bankbot_name = settings.bankbot_name
     @bankbot_deposit_threshold = settings.bankbot_deposit_threshold
     @bankbot_room_id = settings.bankbot_room_id
+    bankbot_enabled = settings.bankbot_enabled
+    @local_currency = town_data[@character_hometown]['currency']
 
     sell_gems("#{settings.gem_pouch_adjective} pouch") if settings.sell_loot_pouch
-
     check_spare_pouch(settings.spare_gem_pouch_container, settings.gem_pouch_adjective) if settings.spare_gem_pouch_container
 
     sell_bundle if settings.sell_loot_bundle
 
-    return if settings.sell_loot_skip_bank
-    exchange_coins
+    return if settings.sell_loot_skip_bank && !bankbot_enabled
+
+    @keep_coppers = parse(settings.sell_loot_money_on_hand.split(' '))
+    @keep_coppers = @bankbot_deposit_threshold if bankbot_enabled
+
+    if bankbot_enabled
+      if @bankbot_name.nil? || @bankbot_room_id.nil?
+        echo "Error. Bankbot_enabled: is set to true, but bankbot_name: and bankbot_room_id are empty."
+        echo "You must define bankbot_name: and bankbot_room_id: in your YAML settings."
+        exit
+      end
+    end
 
-    keep_copper = parse(settings.sell_loot_money_on_hand.split(' '))
-    deposit_coins(keep_copper)
-    check_bankbot(settings.hometown, keep_copper)
+    if settings.sell_loot_skip_bank && bankbot_enabled
+      $CURRENCIES.each{|currency|
+        give_money_to_bankbot(currency, currency == @local_currency ? @keep_coppers : 0)
+      }
+      return
+    end
+
+    exchange_coins
+    if bankbot_enabled == false
```

Comment:
I recommend reversing the order of the conditionals, so the `if true` clause is first.

Reply:
updated